### PR TITLE
Match some register nonmatchings

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

File: src/friend_area_action_menu.c [hunk 1]
```diff
@@ -449,11 +440,11 @@ void sub_80278B4(void)
...
       case FRIEND_AREA_ACTION_MENU_ACTION_MAKE_LEADER:
-        pokeStruct2 = &gRecruitedPokemonRef->pokemon[sUnknown_203B2BC->targetPoke];
+        newLeader = &gRecruitedPokemonRef->pokemon[sUnknown_203B2BC->targetPoke];
         playerStruct = GetPlayerPokemonStruct();
-        if (!pokeStruct2->isTeamLeader) {
+        if (!newLeader->isTeamLeader) {
           playerStruct->isTeamLeader = FALSE;
-          pokeStruct2->isTeamLeader = TRUE;
+          newLeader->isTeamLeader = TRUE;
           nullsub_104();
```
Comment (Kermalis) [inline src/friend_area_action_menu.c:448]: It'd be cool to make a comment at nullsub_104 stating it's called when the team leader is changed by the player. Eventually when we are at documentation, it'll be cool to see what all the nullsubs were used for (probably printing or breakpoints)

File: src/code_8097504.c [hunk 2]
```diff
@@ -4,29 +4,25 @@
...
 #include "event_flag.h"
 #include "code_80A26CC.h"
+#include "code_80972F4.h"
 
 ALIGNED(4) const char gMeetNinetalesText[] =  "Meet Ninetales.";
 ALIGNED(4) const char gAvoidCaptureText[] = "Avoid capture.";
 ALIGNED(4) const char gFinalScenarioText[] = _("Defeat the final Pokémon.");
 
+// Needed to match
+static inline s16 Self(s16 a)
```
Comment (Kermalis) [inline src/code_8097504.c:14]: This smells similar to the `SpeciesId()` inline but instead it's for maze ids

Comment (DizzyEggg) [reply src/code_8097504.c:14]: Yeah, it makes me wonder if `SpeciesId` should actually exist and there's some unholy `InlineCastToS16` used instead

File: src/friend_area_action_menu.c (hunk 1 above)
Comment (DizzyEggg) [reply src/friend_area_action_menu.c:448]: Would be cool to see if the nullsubs are actually used in JP or Blue versions.

File: src/code_8097504.c (hunk 2 above)
Comment (Kermalis) [reply src/code_8097504.c:14]: Nah I think it is definitely a SpeciesId but there also happen to be 2 other times they do that kind of conversion (like maze/friendarea and maybe item/move or something). We'll keep finding them

Comment (DizzyEggg) [reply src/code_8097504.c:14]: True, but there are issues all around the codebase with s16 arguments which don't match. So unless they had conversion used for everything, it's possible they performed some weird inline cast for all s16. 
Though some of them are unknown arguments, so maybe they're indeed just species/moves/items/etc.

**Review by AnonymousRandomPerson — APPROVED** [review]
Will wait for comments to be resolved before merging.

**Review by SethBarberee — APPROVED** [review]

**Pull request merged**